mkosi/backend.py

```diff
@@ -395,50 +395,50 @@ def nspawn_params_for_blockdev_access(args: CommandLineArguments, loopdev: str)
     for partno in (args.esp_partno, args.bios_partno, args.root_partno, args.xbootldr_partno):
         if partno is not None:
             p = partition(loopdev, partno)
-            if os.path.exists(p):
+            if p.exists():
                 params += [f"--bind-ro={p}", f"--property=DeviceAllow={p}"]
 
-    params.extend(f"--setenv={env}" for env in args.environment)
+    params += [f"--setenv={env}" for env in args.environment]
 
     return params
 
 
 def run_workspace_command(
     args: CommandLineArguments,
-    root: str,
-    cmd: List[str],
+    root: Path,
+    cmd: Sequence[PathString],
     network: bool = False,
     env: Optional[Dict[str, str]] = None,
     nspawn_params: Optional[List[str]] = None,
 ) -> None:
-    cmdline = [
+    nspawn = [
         "systemd-nspawn",
         "--quiet",
-        "--directory=" + root,
+        f"--directory={root}",
         "--uuid=" + args.machine_id,
         "--machine=mkosi-" + uuid.uuid4().hex,
         "--as-pid2",
         "--register=no",
-        "--bind=" + var_tmp(root) + ":/var/tmp",
+        f"--bind={var_tmp(root)}:/var/tmp",
         "--setenv=SYSTEMD_OFFLINE=1",
     ]
 
     if network:
         # If we're using the host network namespace, use the same resolver
-        cmdline += ["--bind-ro=/etc/resolv.conf"]
+        nspawn += ["--bind-ro=/etc/resolv.conf"]
     else:
-        cmdline += ["--private-network"]
+        nspawn += ["--private-network"]
 
     if env:
-        cmdline += [f"--setenv={k}={v}" for k, v in env.items()]
+        nspawn += [f"--setenv={k}={v}" for k, v in env.items()]
 
     if nspawn_params:
-        cmdline += nspawn_params
+        nspawn += nspawn_params
 
-    result = run(cmdline + ["--"] + cmd, check=False)
+    result = run([*nspawn, "--", *cmd], check=False)
     if result.returncode != 0:
         if "workspace-command" in ARG_DEBUG:
-            run(cmdline, check=False)
+            run(nspawn, check=False)  # XXX:???
```

Comment:
I don't understand this comment at all

Reply:
I was a note to myself, that I don't understand this part of the code. I dropped it now.